### ISSUE TITLE
(Button) a11y improvements

### ISSUE THREAD

**Issue by MunifTanjim** [optimizely-axiom/optiaxiom#67]
### Disabled State

Disabled button should have `aria-disabled`

Ref: https://css-tricks.com/making-disabled-buttons-more-inclusive/

If button is rendered as anchor, i.e. `<Button asChild><a>Click Me</a></Button>`, the same behavior should be available.

### Keyboard Interaction

Should trigger `onClick` if `Space` or `Enter` is pressed when button is focused.

If button is rendered as anchor, i.e. `<Button asChild><a>Click Me</a></Button>`, the same interaction should be available.

Ref: https://www.w3.org/WAI/ARIA/apg/patterns/button/#:~:text=Keyboard%20Interaction-,When%20the%20button%20has%20focus%3A,%3A%20Activates%20the%20button.,-Following%20button%20activation
